### PR TITLE
Better validation and unit tests  for updateVSSTree

### LINKED ISSUE #191: updateVSSTree broken
https://github.com/eclipse/kuksa.val/commit/3a2cb5b439c9f9269cc0017c55f6cec40b04be4d   broke updateVSSTree 

```
> updateVSSTree ../../kuksa_viss_client/test-roadster-elon.json 
{
  "action": "updateVSSTree",
  "metadata": {
    "Vehicle": {
      "type": "branch",
      "uuid": "ccc825f94139544dbb5f4bfd033bece6",
      "children": {
        "Speed": {
          "max": 9001,
          "type": "sensor",
          "uuid": "efe50798638d55fab18ab7d43cc490e9"
        },
        "Private": {
          "type": "branch",
          "uuid": "4161866b048a5b76aa3124dec82e0260",
          "children": {
            "ThrustersActive": {
              "datatype": "boolean",
              "type": "sensor",
              "uuid": "304b0817a2df524fa442c6a2d2742ed0"
            }
          }
        }
      }
    }
  },
  "requestId": "7252e425-8a66-4111-a0f9-7b9fffbefe0c",
  "error": "timeout"
}
```

fails with ERROR: JSON key not found error, as the expected format was changed.

Need to be fixed and needs a unittest

### PR DESCRIPTION
**Pull request by SebastianSchildt**
Will fix #191, adding unit tests, and some cleanup in schema validation code (less duplication)
I also added two simple tests for updateMetadata (had none so far), which slightly increases line coverage in command processor

### CODE REVIEW

File: src/VssCommandProcessor.cpp [hunk 1]
```diff
@@ -155,17 +155,30 @@ string VssCommandProcessor::processUnsubscribe(const string & request_id,
...
+  } catch (std::exception &e) {
+    logger->Log(LogLevel::ERROR, "Unhandled error: " + string(e.what()));
+    return JsonResponses::malFormedRequest(
+        requestValidator->tryExtractRequestId(request) , "updateVSSTree", string("Unhandled error: ") + e.what());
+  }
+
   jsoncons::json answer;
   answer["action"] = "updateVSSTree";
-  answer["requestId"] = request_id;
+  answer["requestId"] = request["requestId"].as<std::string>();
```
[the remark this answers is not in the document]
Comment (wenwenchenbosch) [reply src/VssCommandProcessor.cpp:176]: but I see in line 227, it was also
```
result["requestId"] = requestId;
```
some quite inconsistent

File: src/VSSRequestValidator.cpp [hunk 2]
```diff
@@ -1,111 +1,102 @@
...
 
-    this->setValidator->validate(request, reporter);
 
-    if (!valid) {
-        throw jsoncons::jsonschema::schema_error("VSS set malformed: "+ss.str());
-    }
+VSSRequestValidator::VSSRequestValidator(std::shared_ptr<ILogger> loggerUtil)  {
+  this->logger = loggerUtil;
+
+  this->getValidator            = new MessageValidator(VSS_JSON::SCHEMA_GET);
```
Comment (wenwenchenbosch) [inline src/VSSRequestValidator.cpp:66]: Maybe a little bit modern c++
```
  VSSRequestValidator::VSSRequestValidator: 
    logger(loggerUtil), 
    getValidator(VSS_JSON::SCHEMA_GET)...{}
```

File: include/VSSRequestValidator.hpp [hunk 3]
```diff
@@ -29,25 +29,25 @@
...
 
     private:
-        std::shared_ptr<jsoncons::jsonschema::json_schema<jsoncons::json> > getSchema;
-        std::shared_ptr<jsoncons::jsonschema::json_schema<jsoncons::json> > setSchema;
-        std::shared_ptr<jsoncons::jsonschema::json_schema<jsoncons::json> > updateTreeSchema;
-        jsoncons::jsonschema::json_validator<jsoncons::json> *getValidator;
-        jsoncons::jsonschema::json_validator<jsoncons::json> *setValidator;
-        jsoncons::jsonschema::json_validator<jsoncons::json> *updateTreeValidator;
+        class MessageValidator;
+        VSSRequestValidator::MessageValidator *getValidator;
```
Comment (wenwenchenbosch) [inline include/VSSRequestValidator.hpp:46]: there pointers are perfekt as unique
```suggestion
        unique_ptr<MessageValidator> getValidator;
```

**Comment by wenwenchenbosch** [comment]
super, VSSRequestValidator is now much simpler, may be a little bitte modern c++ code would be cool :D

**Review by wenwenchenbosch — APPROVED** [review]
